### PR TITLE
P1/feat/46 refactor code

### PR DESCRIPTION
**Pull request by netlightebba**
I will go through this PR in a meeting 31/8 to ease the code review.

This PR will make a lot of changes in the structure of the code, feel free to ask me for help when rebasing your current branches after this PR merges into dev.

### CODE REVIEW

**Review by netlight-saraandersson — APPROVED** [review]

File: app/assets/main/react_components/User/CarouselCategoryButton.jsx [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
+import { useUserActions } from "./contexts/UserActionsContext";
 
 const CarouselCategoryButton = ({
   categoryName,
   categoryID,
-  setCategory,
   active,
   setAllCategories,
 }) => {
+  const updateCategory = useCategoryUpdate();
```
Comment (netlight-johannadagfalk) [inline app/assets/main/react_components/User/CarouselCategoryButton.jsx:15]: Should this be named setCategory to follow the same structure? :)

File: app/assets/main/react_components/User/ClimateActionsContainer.jsx [hunk 2]
```diff
@@ -1,147 +1,44 @@
-import React, { useState, useEffect } from "react";
+import React, { useEffect, useState } from "react";
 import CarouselContainer from "./CarouselContainer.jsx";
-import KanbanActionContainer from "./KanbanActionContainer.jsx";
+//import KanbanActionContainer from "./KanbanActionContainer.jsx";
```
Comment (elinaolsson) [inline app/assets/main/react_components/User/ClimateActionsContainer.jsx:3]: Remove?

File: app/assets/main/react_components/User/ClimateActionsContainer.jsx [hunk 3]
```diff
@@ -1,147 +1,44 @@
...
 import CarouselActionItem from "./CarouselActionItem.jsx";
-import Card from "./Card.jsx";
+import { useDeletedAction } from "./contexts/DeletedActionContext.js";
+import {
+  useClimateActions,
+  useClimateActionsUpdate,
+  useClimateActionsOriginal,
+} from "./contexts/ClimateActionsContext.js";
+//import Card from "./Card.jsx";
```
Comment (elinaolsson) [inline app/assets/main/react_components/User/ClimateActionsContainer.jsx:11]: Remove?

File: app/assets/main/react_components/User/CarouselCategoryButton.jsx [hunk 4]
```diff
@@ -20,6 +30,28 @@ const CarouselCategoryButton = ({
     setAllCategories(false);
   };
 
+  const setCategory = (cat) => {
+    //***** HERE WE UPDATE THE CONTEXT */
```
Comment (netlight-johannadagfalk) [inline app/assets/main/react_components/User/CarouselCategoryButton.jsx:34]: I see the reason why updateCategory wasnt named setCategory... Perhaps change them to the opposite?

File: app/assets/main/react_components/User/ClimateActionsContainer.jsx [hunk 5]
```diff
@@ -155,31 +52,20 @@ const ClimateActionsContainer = ({
...
-      <Sidebar
-        setLocalAccepted={setLocalAccepted}
-        columns={columns}
-        setColumns={setColumns}
-        setTotUserActions={setTotUserActions}
         categoryColor={categoryColor}
       />
+
+      <Sidebar categoryColor={categoryColor} />
+      {/* <KanbanActionContainer /> */}
```
Comment (elinaolsson) [inline app/assets/main/react_components/User/ClimateActionsContainer.jsx:68]: Remove?

File: app/assets/main/react_components/User/contexts/UserActionsContext.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
+import React, { useState, useContext } from "react";
+
+//*** A context is used to share data that can be considered as "global" for the react tree ***/
```
Comment (elinaolsson) [inline app/assets/main/react_components/User/contexts/UserActionsContext.js:3]: Nice documentation! (ongoingly)

**Review by elinaolsson — APPROVED** [review]
You rock 👊

**Review by netlight-johannadagfalk — APPROVED** [review]

**Pull request merged**